### ISSUE TITLE
server freezing - getcubes

### ISSUE THREAD

**Issue by PhanaticD** [ProjectKorra/ProjectKorra#612]
http://pastebin.com/xMgd7tDy

My server has crashed a few times with this, I believe it is due to setting excessive velocities after doing some more research and looking at people who have had similar issues

to fix this I THINK this might be needed, not completely sure, I am testing it out right now on my server

in GeneralMethods.setVelocity
```
        if(velocity.getX() > 4){
        	velocity.setX(4);
        }
        if(velocity.getX() < -4){
        	velocity.setX(-4);
        }
        if(velocity.getY() > 4){
        	velocity.setY(4);
        }
        if(velocity.getY() < -4){
        	velocity.setY(-4);
        }
        if(velocity.getZ() > 4){
        	velocity.setZ(4);
        }
        if(velocity.getZ() < -4){
        	velocity.setZ(-4);
        }
```

**Comment by SteveTheDog** [comment]
Can vouch that I have this issue all the time too, and I came to the same conclusion.

**Comment by PhanaticD** [comment]
I have not had a freeze yet, whereas I previously was getting one every hour or so

**Comment by Sobki** [comment]
Steve, can you please PM me on ProjectKorra forums so I can try get more information about it. Same with you, Phanatic. I'll do some digging with this and try find a solution

**Comment by RoboMWM** [comment]
I know @ichbinjoe looked into this and implemented a velocity warning in paper. However, I have yet to receive this issue despite having and using plugins that set excessive velocity ever since this was added. (I did use to rarely encounter the getCubes crash before though.)

**Comment by Ichbinjoe** [comment]
I've been summoned!

From what I remember, this explanation is right on the money. I was able to get hangs when setting the velocity +1M in a few axis. It does take quite a bit to get it to hang and die. I chose to implement the warning vs simply capping off to keep behavioral backwards comparability (even though it may still crash the server).

I think the original issue is due to two scenarios:

1) Developers using stupid high velocities.
2) Developers using equations such as an inverse which approach infinity. That just sounds bad.

Hope that helps!
